optaplanner-core/src/main/java/org/optaplanner/core/impl/score/director/AbstractScoreDirector.java

```diff
@@ -776,6 +784,16 @@ protected String buildScoreCorruptionAnalysis(InnerScoreDirector<Solution_, Scor
         return constraintMatchMap;
     }
 
+    protected boolean isConstraintConfiguration(Object problemFactOrEntity) {
+        SolutionDescriptor<Solution_> solutionDescriptor = scoreDirectorFactory.getSolutionDescriptor();
+        MemberAccessor constraintConfigurationAccessor = solutionDescriptor.getConstraintConfigurationMemberAccessor();
+        if (constraintConfigurationAccessor == null) {
+            return false;
+        }
+        Object constraintConfiguration = constraintConfigurationAccessor.executeGetter(workingSolution);
```

Comment:
Intresting. I would have just checked if it's an instance of that ConstraintConfiguration class. No idea which is better:

Proposal A) check executeGetter() returns same instance
Pro: they have other problem facts that are instances of that class.
Con: every single after/before call in addPFC (sometimes n per PFC) calls that executeGetter.

Proposal B) check instanceOf
This presumes that normally a dataset doesn't have other problem facts that implement that constraint configuration class. I 've never seen that happen so far.

Proposal C) Do A), but cache the constraintConfiguration instance on scoreDirector.
Con: more work/maintenance/complexity